blocks/api/test/parser.js

```diff
@@ -282,5 +282,66 @@ describe( 'block parser', () => {
 				'core/test-block', 'core/void-block',
 			] );
 		} );
+
+		it( 'should parse primitive blocks like nextpage', () => {
+			registerBlockType( 'core/nextpage', {
+				save() {
+					return null;
+				},
+			} );
+			const parsed = parse(
+				'<!--nextpage-->'
+			);
+
+			expect( parsed ).to.have.lengthOf( 1 );
+			expect( parsed.map( ( { name } ) => name ) ).to.eql( [ 'core/nextpage' ] );
+		} );
+
+		it( 'should parse the post content, including primitive blocks in it', () => {
+			registerBlockType( 'core/test-block', {
+				save() {
+					return null;
+				},
+			} );
+			registerBlockType( 'core/nextpage', {
+				save() {
+					return null;
+				},
+			} );
+			registerBlockType( 'core/unknown-block', {
+				// Currently this is the only way to test block content parsing?
+				attributes: function( rawContent ) {
+					return {
+						content: rawContent,
+					};
+				},
+
+				save() {
+					return null;
+				},
+			} );
+
+			setUnknownTypeHandler( 'core/unknown-block' );
+
+			const parsed = parse(
+				'<p>Cauliflower</p>' +
+				'<!-- wp:core/test-block -->Ribs<!-- /wp:core/test-block -->' +
+				'\n<p>Broccoli</p>\n' +
+				'<!--nextpage-->' +
+				'<p>Romanesco</p>'
+			);
+
+			expect( parsed ).to.have.lengthOf( 5 );
+			expect( parsed.map( ( { name } ) => name ) ).to.eql( [
+				'core/unknown-block',
+				'core/test-block',
+				'core/unknown-block',
+				'core/nextpage',
+				'core/unknown-block',
+			] );
+			expect( parsed[ 0 ].attributes.content ).to.eql( '<p>Cauliflower</p>' );
+			expect( parsed[ 2 ].attributes.content ).to.eql( '<p>Broccoli</p>' );
+			expect( parsed[ 4 ].attributes.content ).to.eql( '<p>Romanesco</p>' );
+		} );
```

Comment:
This last test case in particular is probably a better fit for a new file in `blocks/test/fixtures/`, for example `misc__multi-blocks-and-nextpage.html`.  This way it will be both easier to set up and more thoroughly tested.